File: codingjs/fibonacci.js
function fibonacci(n: number): number {
  return fibonacciHelper(0, 1, n);
}
function fibonacciHelper(f1: number, f2: number, n: number): number {
  if (n === 0) return f1;
  return fibonacciHelper(f2, f1 + f2, --n);
}

File: codingbat/fibonacci.java
public int fibonacciHelper(int n, int s1, int s2) {
  if (n == 0) return s1;
  return fibonacciHelper(--n, s2, s1 + s2);
}

public int fibonacci(int n) {
  return fibonacciHelper(n, 0, 1);
}

